src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2619,6 +2619,111 @@ def import_on_boot(self, job):
             self.dismissed_import_disk_jobs.add(current_import_job["id"])
 
 
+class PoolDatasetUserPropService(CRUDService):
+
+    class Config:
+        namespace = 'pool.dataset.userprop'
+
+    @filterable
+    async def query(self, filters=None, options=None):
```

Comment:
Do you think this should be a method, given that filter_list may block, not being a coroutine?